js/calc.js

```diff
@@ -0,0 +1,102 @@
+//FUNCTIONS
+function makeButtonNum(newButton, index, val){
+  newButton = numbers[index];
+  newButton.addEventListener('click',handleClick);
+  function handleClick(){
+    line += val.toString();
+    changeDisplay(line);
+  }
+  return newButton;
+}
+
+function makeOperatorButton(newButton, index, operator){
+  newButton = operators[index];
+  newButton.addEventListener('click',handleClick);
+  function handleClick(){
+    line += operator;
+    changeDisplay(line);
+  }
+}
+
+function clearLine(){
+  line = '';
+  changeDisplay(line);
+}
+
+function equalsLine(){
+  //keep a runnign total
+  var runningTotal=0;
+  //Split the line into an array whenever it reads an operator, use a regex for this
+  var linecopy = line;
+  //((/[*/+-]/) gives numbers
+  arrayNumbers = line.split(/[*/+-]/);
```

Comment:
Very cool and smart solution to the problem - but perhaps a bit overcomplicated. Don't be afraid to keep things simple!